lib/Doctrine/ORM/Persisters/Entity/JoinedSubclassPersister.php

```diff
@@ -487,7 +487,7 @@ protected function getInsertColumnList(): array
 
         // Add discriminator column if it is the topmost class.
         if ($this->class->name === $this->class->rootEntityName) {
-            $columns[] = $this->class->getDiscriminatorColumn()['name'];
+            $columns[] = $this->class->getDiscriminatorColumn()->name;
```

Comment:
This change and the one below make static analysis issues go away, that's why I decided to add them. There will be PRs dedicated to migrating to the object interface later.